src/trails.erl

```diff
@@ -1,35 +1,78 @@
 -module(trails).
 
-
 -export([single_host_compile/1]).
 -export([compile/1]).
 -export([trail/2]).
 -export([trail/3]).
 -export([trail/4]).
+-export([trail/5]).
+
+-opaque trail() ::
+  #{ path_match  => integer()
+   , constraints => integer()
+   , handler     => module()
+   , options     => any()
+   , metadata    => map()
+   }.
+-export_type([trail/0]).
+
+-type trails() :: [ trails:trail() | cowboy_router:route_path() ].
```

Comment:
I think it doesn't work since `cowboy_router:route_path()` includes the host part that `trails:trail()` does not.